api/v1beta1/cluster_types.go

```diff
@@ -860,7 +865,17 @@ func (c *Cluster) GetClassKey() types.NamespacedName {
 	if c.Spec.Topology == nil {
 		return types.NamespacedName{}
 	}
-	return types.NamespacedName{Namespace: c.GetNamespace(), Name: c.Spec.Topology.Class}
+
+	return types.NamespacedName{Namespace: c.GetInfrastructureNamespace(), Name: c.Spec.Topology.Class}
+}
+
+// GetInfrastructureNamespace returns common namespace for the cluster infrastructure.
+func (c *Cluster) GetInfrastructureNamespace() string {
```

Comment:
Considering that this func looks at  c.Spec.Topology and not at c.Spec.InfrastructureRef, I'm a little bit confused by "Infrastructure" in the name.

Also, if possible, I would prefer to try to keep API package as clean as possible not use it as a way to share util between packages (I know we did this in the past in few cases, let's try to not add more).
E.g. we can thing about something under internal/topology, or also duplicate this if could work for me